cache/lru_cache.cc

```diff
@@ -418,18 +402,16 @@ void LRUCacheShard::Erase(const Slice& key, uint32_t hash) {
     MutexLock l(&mutex_);
     e = table_.Remove(key, hash);
     if (e != nullptr) {
-      last_reference = Unref(e);
-      if (last_reference) {
+      e->SetInCache(false);
+      if (!e->HasRefs()) {
         usage_ -= e->charge;
-      }
-      if (last_reference && e->InCache()) {
         LRU_Remove(e);
+        last_reference = true;
```

Comment:
@siying but it IS in LRU list. Because it IS in table_ and it HAS NO refs. Hence it is in LRU.
There is no need to check for InCache(), since it's TRUE anyway (element is in table_).

Reply:
OK. Add an assert and a comment explaining why it must be there. I'll be OK with it.